### PR TITLE
Bug 1338334 - fix analyze_failures.py when we have empty fixed_by_com…

### PR DESCRIPTION
**Pull request by jmaher**
…mit arrays

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/2153)

### CODE REVIEW

File: treeherder/seta/analyze_failures.py [hunk 1]
```diff
@@ -59,8 +59,16 @@ def get_failures_fixed_by_commit():
     failures = {}
     # We're assuming that sheriffs always anotate failed jobs correctly using "fixed by commit"
     for job_note in models.JobNote.objects.filter(failure_classification=2):
```
Comment (rwood-moz) [inline treeherder/seta/analyze_failures.py]: How about we try to fix Bug 1338570 now also? I *believe* you'd just add:

for job_note in models.JobNote.objects.filter(failure_classification=2).select_related('job'):

File: tests/seta/conftest.py [hunk 2]
```diff
@@ -116,22 +116,56 @@ def jp_index_fixture(job_priority_list):
...
+    counter = 0
     for job in Job.objects.all():
-        for failure_classification_id in [2, 3]:
+        counter += 1
+
+        # skip first 5 jobs related to the man-o-lantern
+        if counter < 6:
+            JobNote.objects.create(job=job,
+                                    failure_classification_id=2,
+                                    user=test_user, text="you look like a man-o-lantern")
```
Comment (rwood-moz) [inline tests/seta/conftest.py]: just a nit, two extra spaces on indent here two lines above

File: tests/seta/conftest.py [hunk 3]
```diff
@@ -116,22 +116,56 @@ def jp_index_fixture(job_priority_list):
...
+
+        # add 5 jobs with raw revision, expected: 31415926535
+        if counter < 11:
+            JobNote.objects.create(job=job,
+                                   failure_classification_id=2,
+                                   user=test_user, text="314159265358")
+            continue
+
+        # Add 5 jobs with full url to revision, expected to map to 31415926535
+        if counter < 14:
```
Comment (rwood-moz) [inline tests/seta/conftest.py]: Did you intend to only add 4 jobs (I'm guessing the comment is wrong)

File: tests/seta/conftest.py [hunk 4]
```diff
@@ -116,22 +116,56 @@ def jp_index_fixture(job_priority_list):
...
+                                   user=test_user, text="http://hg.mozilla.org/mozilla-central/314159265358")
+            continue
+
+        # Add one job that doesn't parse properly and produces a revision with no data
+        if counter < 15:
             JobNote.objects.create(job=job,
-                                   failure_classification_id=failure_classification_id,
-                                   user=test_user, text="you look like a man-o-lantern")
+                                    failure_classification_id=2,
+                                    user=test_user, text="call me empty")
```
Comment (rwood-moz) [inline tests/seta/conftest.py]: nit, extra spaces in the indent for two lines

File: tests/seta/conftest.py [hunk 5]
```diff
@@ -116,22 +116,56 @@ def jp_index_fixture(job_priority_list):
...
+            continue
+
+        # Add one job with trailing slash, expected to map to 31415926535
+        JobNote.objects.create(job=job,
```
Comment (rwood-moz) [inline tests/seta/conftest.py]: Since there's no 'if counter..' for this one, if there happens to be more than 15 (Job.objects) then this could potentially add more than just one more here right? How do you know for certain there are only 15 items in Jobs.objects.all()?

File: tests/seta/conftest.py [hunk 6]
```diff
@@ -116,22 +116,56 @@ def jp_index_fixture(job_priority_list):
...
+                                failure_classification_id=2,
+                                   user=test_user, text="http://hg.mozilla.org/mozilla-central/314159265358/")
```
Comment (rwood-moz) [inline tests/seta/conftest.py]: nit, extra spaces in the indentations

File: treeherder/seta/analyze_failures.py [hunk 7]
```diff
@@ -59,8 +59,16 @@ def get_failures_fixed_by_commit():
...
+        # prevent an empty string
+        if not job_note.text:
+            continue
```
Comment (rwood-moz) [inline treeherder/seta/analyze_failures.py]: maybe create another job_note with an empty text string, to test this too?

File: treeherder/seta/analyze_failures.py (hunk 1 above)
Comment (edmorley) [inline treeherder/seta/analyze_failures.py]: That adds the join to the jobs table, which is a start, but needs the others too. In the other bug (about slow API, but root cause is this same line) I gave Armen an example of how to fix it:
https://bugzilla.mozilla.org/show_bug.cgi?id=1330354#c5

**Comment by jmaher** [comment]
thanks for all the feedback, for the select_related, I want to handle that in bug https://bugzilla.mozilla.org/show_bug.cgi?id=1330354, otherwise added a couple extra tests here and cleaned up some whitespace.
